routes/chat.js

```diff
@@ -118,9 +80,45 @@ router.post('/chat', async function (req, res) {
       chat: chatID,
     });
 
-    res.json(chat);
+    res.status(201).json(chat);
   } else {
     // join
-    res.status(204).json(commonChats);
+    res.status(200).json(commonChats);
```

Comment:
commonChats includes also the groups that the two users share. To get the chat between those two users, you need to check that the chats only has two users.
However, for now this works enough. Maybe it's better to do `commonChats[0]` so we return the single chat.